src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -79,9 +77,14 @@ const postCompaniesHouseSearch = async (req: Request, res: Response) => {
     };
 
     // checks if input is correctly formatted before searching
-    const validationErrors = companiesHouseValidation(body);
+    const response = await companiesHouseSearch(body);
+    const { validationErrors, error, company } = response;
 
-    if (validationErrors) {
+    if (error) {
+      return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
+    }
+
+    if (Object.keys(validationErrors).length) {
```

Comment:
Can we rename `companyDetailsTemplate` to `TEMPLATE` so that it's consistent with other controllers

Reply:
Does this work if `validationErrors` is not returned from the response?

Reply:
done

Reply:
validation errors will always be returned from the function.  Either as an empty object or as errors.  Will add an extra fails safe case `    if (validationErrors && Object.keys(validationErrors).length) {`